docs/data/data-grid/state/RestoreStateApiRef.tsx

```diff
@@ -0,0 +1,324 @@
+  | { type: 'closePopper' };
+
+const demoReducer: React.Reducer<DemoState, DemoActions> = (state, action) => {
+  switch (action.type) {
+    case 'createView': {
+      const id = Math.random().toString();
+
+      return {
+        ...state,
+        activeViewId: id,
+        newViewLabel: '',
+        views: {
+          ...state.views,
+          [id]: { label: state.newViewLabel, value: action.value },
+        },
+      };
+    }
+
+    case 'deleteView': {
+      const views = Object.fromEntries(
+        Object.entries(state.views).filter(([id]) => id !== action.id),
+      );
+
+      let activeViewId: string | null;
+      if (state.activeViewId !== action.id) {
+        activeViewId = state.activeViewId;
+      } else {
+        const viewIds = Object.keys(state.views);
+
+        if (viewIds.length === 0) {
+          activeViewId = null;
+        } else {
+          activeViewId = viewIds[0];
+        }
+      }
+
+      return {
+        ...state,
+        views,
+        activeViewId,
+      };
+    }
+
+    case 'setActiveView': {
+      return {
+        ...state,
+        activeViewId: action.id,
+        isMenuOpened: false,
+      };
+    }
+
+    case 'setNewViewLabel': {
+      return {
+        ...state,
+        newViewLabel: action.label,
+      };
+    }
+
+    case 'togglePopper': {
+      return {
+        ...state,
+        isMenuOpened: !state.isMenuOpened,
+        menuAnchorEl: action.element,
+      };
+    }
+
+    case 'closePopper': {
+      return {
+        ...state,
+        isMenuOpened: false,
+      };
+    }
+
+    default: {
+      return state;
+    }
+  }
+};
+
+const DEMO_INITIAL_STATE: DemoState = {
+  views: {},
+  newViewLabel: '',
+  isMenuOpened: false,
+  menuAnchorEl: null,
+  activeViewId: null,
+};
+
+const ViewListItem = (props: {
+  view: StateView;
+  viewId: string;
+  selected: boolean;
+  onDelete: (viewId: string) => void;
+  onSelect: (viewId: string) => void;
+}) => {
+  const { view, viewId, selected, onDelete, onSelect } = props;
+
+  return (
+    <ListItem
+      disablePadding
+      selected={selected}
+      onClick={() => onSelect(viewId)}
+      secondaryAction={
+        <IconButton
+          edge="end"
+          aria-label="delete"
+          size="small"
+          onClick={(event) => {
+            event.stopPropagation();
+            onDelete(viewId);
+          }}
+        >
+          <DeleteIcon />
+        </IconButton>
+      }
+    >
+      <ListItemButton>
+        <ListItemText primary={view.label} />
+      </ListItemButton>
+    </ListItem>
+  );
+};
+
+const NewViewListButton = (props: {
+  label: string;
+  onLabelChange: (
+    e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+  ) => void;
+  onSubmit: () => void;
+  isValid: boolean;
+}) => {
+  const { label, onLabelChange, onSubmit, isValid } = props;
+  const [isAddingView, setIsAddingView] = React.useState(false);
+
+  if (isAddingView) {
+    return (
+      <React.Fragment>
```

Comment:
It doesn't look good. The text field is too tall. I would use `prompt` or put it inside a `Dialog`.

![image](https://user-images.githubusercontent.com/42154031/159267986-a7f5f085-9931-43a4-a4e1-efde18f2ab23.png)